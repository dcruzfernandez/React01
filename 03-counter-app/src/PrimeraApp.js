import PropTypes from 'prop-types'

export const PrimeraApp = ({saludo,subtitulo}) => {
    
    return (
        <>
          <h1>{saludo}!!</h1>
          {/* <pre>{JSON.stringify(saludo,null,3)}</pre>   */}
          <p>{subtitulo}</p>
        </>
    )
}
//definir tipos de propiedades
PrimeraApp.propTypes={
    saludo:PropTypes.string.isRequired
}

//definir valores predeterminados para las propiedades
PrimeraApp.defaultProps={
    subtitulo:'Soy subtitulo'
}
