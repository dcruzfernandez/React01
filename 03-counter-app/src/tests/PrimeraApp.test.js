//import { render } from "@testing-library/react";
import { PrimeraApp } from "../PrimeraApp";
import '@testing-library/jest-dom';
import {shallow} from 'enzyme'

describe('Pruebas de componente PrimeraApp', () => {
    // test('Prueba 1 debe mostrar el mensaje Hola Mundo', () => {
    //     const saludo='Hola Mundo';
        
    //     const {getByText}=render(<PrimeraApp saludo={saludo}/>);
    //     console.log(getByText);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // });

    test('Debe mostrar PrimeraApp correctamente', () => {
       const wrapper=shallow(<PrimeraApp saludo='Hola'/>) 

       expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe mostrar PrimeraApp con subtitulo', () => {

        const saludo='hola';

        const subsaludo='adios';
        const wrapper=shallow(<PrimeraApp saludo={saludo} subtitulo={subsaludo}/>); 

        const textoParrafo=wrapper.find('p').text();
        
        console.log(textoParrafo);

        expect(textoParrafo).toBe(subsaludo);
        
     });

});
