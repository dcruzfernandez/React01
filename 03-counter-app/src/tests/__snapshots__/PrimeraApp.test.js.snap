// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de componente PrimeraApp Debe mostrar PrimeraApp correctamente 1`] = `
<Fragment>
  <h1>
    Hola
    !!
  </h1>
  <p>
    Soy subtitulo
  </p>
</Fragment>
`;
